### PR TITLE
🚑️ Remove circular dependency, fixes #55, #60

### LINKED ISSUE #55: Current master can't do bundle install
```
❯ bundle install
Fetching bundler 2.1.4
Installing bundler 2.1.4
Fetching gem metadata from http://rubygems.org/.......
Resolving dependencies...
Your bundle requires gems that depend on each other, creating an infinite loop. Please remove gem 'os' and try again.
```

### PR DESCRIPTION
**Pull request by pboling**
Fixes #55 & #60 

Signed-off-by: Peter Boling <peter.boling@gmail.com>

### CODE REVIEW

**Comment by pboling** [comment]
@rdp Ping!

File: os.gemspec [hunk 1]
```diff
@@ -46,18 +46,15 @@ Gem::Specification.new do |s|
     s.specification_version = 4
 
     if Gem::Version.new(Gem::VERSION) >= Gem::Version.new('1.2.0') then
-      s.add_runtime_dependency(%q<os>.freeze, [">= 0"])
```
Comment (pboling) [inline os.gemspec:49]: The os gem can't depend on itself.  I think this is part of the messy legacy of `jeweler`.

File: Gemfile.lock [hunk 2]
```diff
@@ -1,25 +1,38 @@
...
 PLATFORMS
-  ruby
-  x86-mingw32
+  arm64-darwin-21
 
 DEPENDENCIES
+  os!
   rake (~> 0.8)
-  rspec (~> 2.5.0)
+  rspec (>= 2.0)
```
Comment (pboling) [inline Gemfile.lock:34]: Loosening here will allow using the `transpec` tool to automatically upgrade the RSpec syntax to RSpec 3+.

**Review by pboling — COMMENTED** [review]
@rdp

**Comment by rdp** [comment]
Do the specs work with 3+ ?

**Comment by pboling** [comment]
@rdp I'll check.  If not I'll convert them to RSpec 3 syntax in this PR as well.

**Comment by pboling** [comment]
@rdp I have fixed the issues with the test suite.  In order to delay the upgrade to a newer RSpec I had to constrain the gem version dependencies in the `gemspec` to the same patch level as what had been used in the last "known good" state, which I assumed was in the `Gemfile.lock`.

This branch now works for both `bundle install`, and `bundle exec rspec`, and all tests pass.  There were two bugs in the test suite that have been fixed.

File: spec/os_spec.rb [hunk 3]
```diff
@@ -144,18 +145,19 @@
 
   it "should provide a path to directory for application config" do
     ENV.stub(:[])
+    home = '/home/user'
 
     if OS.mac?
-      ENV.stub(:[]).with('HOME').and_return('/home/user')
-      assert OS.app_config_path('appname') == '/home/xdg/Library/Application Support/appname'
```
Comment (pboling) [inline spec/os_spec.rb:150]: It's clear you weren't running this on a Mac 😄

File: spec/os_spec.rb [hunk 4]
```diff
@@ -121,7 +121,8 @@
   it "has working cpu count method" do
     cpu_count = OS.cpu_count
     assert cpu_count >= 1
-    assert (cpu_count & (cpu_count - 1)) == 0 # CPU count is normally a power of 2
```
Comment (pboling) [inline spec/os_spec.rb:124]: I use a 6-core, 12 vCPU Macbook Pro.  The power of 2 idea is no longer reasonable.

File: os.gemspec (hunk 1 above)
Comment (rdp) [reply os.gemspec:49]: Yeah I think jeweler was auto inserting this, scary...

**Comment by rdp** [comment]
OK is there a change in this PR?  Or is it the following PR's?  Github doesn't show it very clearly LOL.

**Comment by pboling** [comment]
@rdp For some reason I do not have a merge option. Is a review required for a merge option to appear?

**Pull request merged**